tests/utils/host_utils.cpp

```diff
@@ -906,6 +906,28 @@ int compare_floats(void *a, void *b, int len, double epsilon, QudaPrecision prec
     return compareFloats((float *)a, (float *)b, len, epsilon);
 }
 
+template <typename Float> static double compareFloats_v2(Float *a, Float *b, int len, double epsilon)
+{
+  double global_diff = 0.0;
+  for (int i = 0; i < len; i++) {
+    double diff = fabs(a[i] - b[i]);
+    if (diff > epsilon || std::isnan(diff)) {
+      //printfQuda("ERROR: i=%d, a[%d]=%f, b[%d]=%f\n", i, i, a[i], i, b[i]);
+      return diff;
+    }
+    global_diff = std::max(global_diff, diff);
+  }
+  return global_diff;
+}
+
+double compare_floats_v2(void *a, void *b, int len, double epsilon, QudaPrecision precision)
```

Comment:
Why is a `v2` needed?  Add some comments to explain what `v2` does different from the original.

Reply:
it returns computed deviation rather then an int flag.

Reply:
Ok, can you add a comment on this difference in the code then?